rest/tests/test_rest.py

```diff
@@ -576,3 +585,566 @@ def test_root(self):
+            authors
+        )
+        self.assertEqual(
+            len(list_users_in_repo(self.repo, BaseGroupTypes.CURATORS)),
+            curators
+        )
+        self.assertEqual(
+            len(list_users_in_repo(self.repo, BaseGroupTypes.ADMINISTRATORS)),
+            admins
+        )
+
+    def test_members_get(self):
+        """
+        Members GET
+        """
+        all_members = sorted(
+            [
+                {u'username': u'admin_user', u'group_type': u'administrators'},
+                {u'username': u'curator_user', u'group_type': u'curators'},
+                {u'username': u'author_user', u'group_type': u'authors'}
+            ],
+            key=itemgetter('username')
+        )
+        author_member = [{u'group_type': u'authors'}]
+        curator_member = [{u'group_type': u'curators'}]
+        admin_member = [{u'group_type': u'administrators'}]
+        author_user_member = [{u'username': u'author_user'}]
+        curator_user_member = [{u'username': u'curator_user'}]
+        admin_user_member = [{u'username': u'admin_user'}]
+        # Populate repo groups.
+        self.add_users_to_repo()
+        # all kind of users will get the same results
+        for repo_user in self.all_user_repo:
+            self.login(repo_user.username)
+            # get all the users for all the groups
+            resp_dict = self.get_members(
+                urlfor='base',
+                repo_slug=self.repo.slug
+            )
+            self.assertListEqual(
+                sorted(
+                    resp_dict['results'],
+                    key=itemgetter('username')
+                ),
+                all_members
+            )
+            # get all the groups for specific users
+            resp_dict = self.get_members(
+                urlfor='users',
+                repo_slug=self.repo.slug,
+                username=self.author_user.username
+            )
+            self.assertListEqual(resp_dict['results'], author_member)
+            resp_dict = self.get_members(
+                urlfor='users',
+                repo_slug=self.repo.slug,
+                username=self.curator_user.username
+            )
+            self.assertListEqual(resp_dict['results'], curator_member)
+            resp_dict = self.get_members(
+                urlfor='users',
+                repo_slug=self.repo.slug,
+                username=self.admin_user.username
+            )
+            self.assertListEqual(resp_dict['results'], admin_member)
+            # get one group for a specific users
+            # author
+            resp_dict = self.get_members(
+                urlfor='users',
+                repo_slug=self.repo.slug,
+                username=self.author_user.username,
+                group_type=BaseGroupTypes.AUTHORS
+            )
+            self.assertEqual(resp_dict, author_member[0])
+            # OPTION request for the following 2 test returns 200 (is it ok?)
```

Comment:
A 200 here seems reasonable. It's just validating the the URL is valid. The 404 returned later is artificial, masking user error to prevent information leakage.